### ISSUE TITLE
unsymlinking causes test failures

### ISSUE THREAD

**Issue by kiwifb** [cschwan/sage-on-gentoo#542]
Recently moved to a new profile and migrated to the new structure where `/lib` and `/usr/lib` are folders for 32bits objects and not a link to `lib64` on amd64 system anymore. Because sage's cython code explicitly list `/usr/lib` for compiling code we get a lot of
```
 sage -t --long /usr/lib64/python2.7/site-packages/sage/parallel/decorate.py
**********************************************************************
File "/usr/lib64/python2.7/site-packages/sage/parallel/decorate.py", line 560, in sage.parallel.decorate.fork
Failed example:
    cython('def f(): print(<char*>0)')
Expected nothing
Got:
    /usr/lib/gcc/x86_64-pc-linux-gnu/9.1.0/../../../../x86_64-pc-linux-gnu/bin/ld: skipping incompatible /usr/lib/libm.so when searching for -lm
    /usr/lib/gcc/x86_64-pc-linux-gnu/9.1.0/../../../../x86_64-pc-linux-gnu/bin/ld: skipping incompatible /usr/lib/libm.a when searching for -lm
    /usr/lib/gcc/x86_64-pc-linux-gnu/9.1.0/../../../../x86_64-pc-linux-gnu/bin/ld: skipping incompatible /usr/lib/libpthread.so when searching for -lpthread
    /usr/lib/gcc/x86_64-pc-linux-gnu/9.1.0/../../../../x86_64-pc-linux-gnu/bin/ld: skipping incompatible /usr/lib/libpthread.a when searching for -lpthread
    /usr/lib/gcc/x86_64-pc-linux-gnu/9.1.0/../../../../x86_64-pc-linux-gnu/bin/ld: skipping incompatible /usr/lib/libc.so when searching for -lc
    /usr/lib/gcc/x86_64-pc-linux-gnu/9.1.0/../../../../x86_64-pc-linux-gnu/bin/ld: skipping incompatible /usr/lib/libc.a when searching for -lc
**********************************************************************
1 item had failures:
   1 of  18 in sage.parallel.decorate.fork
    [90 tests, 1 failure, 14.57 s]
----------------------------------------------------------------------
sage -t --long /usr/lib64/python2.7/site-packages/sage/parallel/decorate.py  # 1 doctest failed
----------------------------------------------------------------------
```
Anything that compiles code may send this warning.

**Comment by kiwifb** [comment]
Prefix is probably unaffected because the prefix paths are searched first.

**Comment by kiwifb** [comment]
This is probably the source of all these failures https://github.com/sagemath/sage/blob/develop/src/sage/misc/cython.py#L52

**Comment by kiwifb** [comment]
Source of all the new failures confirmed as being the line above.

**Comment by kiwifb** [comment]
fixed by 589c486d50e615529da05ed2590a00605749392c
